### PR TITLE
[10.x] Add ArrayAccess to Stringable

### PR DESCRIPTION
**Pull request by mathieutu**
Hi,

This little PR adds `ArrayAccess` to `Stringable`.

It allows to make its behavior more like a real string, by proxifying the calls.

```diff
- str('My StRinG')->lower()->value()[0]
+ str('My StRinG')->lower()[0]
```

Thanks,
Matt'

### CODE REVIEW

File: src/Illuminate/Support/Stringable.php [hunk 1]
```diff
@@ -396,7 +397,6 @@ public function lower()
     /**
      * Convert GitHub flavored Markdown into HTML.
      *
-     * @param  array  $options
```
Comment (rodrigopedra) [inline src/Illuminate/Support/Stringable.php:399]: Why this was removed? (and 4 other `@param` and one `@return` at `jsonSerialize`)

File: tests/Support/SupportStringableTest.php [hunk 2]
```diff
@@ -1179,4 +1179,13 @@ public function testToDateThrowsException()
 
         $this->stringable('not a date')->toDate();
     }
+
+    public function testArrayAccess()
+    {
+        $str = $this->stringable('my string');
+        $this->assertSame('m', $str[0]);
+        $this->assertSame('s', $str[4]);
```
Comment (rodrigopedra) [inline tests/Support/SupportStringableTest.php:1187]: This should be `$this->assertSame('t', $str[4]);` or `$this->assertSame('s', $str[3]);`.

|  index |  char |
|:-------:|:-----:|
| 0 | m |
| 1 | y |
| 2 | SPACE |
| 3 | s |
| 4 | t |
| 5 | r |
| 6 | i |
| 7 | n |
| 8 | g |

File: src/Illuminate/Support/Stringable.php (hunk 1 above)
Comment (mathieutu) [reply src/Illuminate/Support/Stringable.php:399]: Hum weird indeed, I wrote that directly in github.dev, the ide should have removed them.

File: tests/Support/SupportStringableTest.php (hunk 2 above)
Comment (mathieutu) [reply tests/Support/SupportStringableTest.php:1187]: Indeed, thanks for the catch.

**Pull request merged**